resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -524,7 +533,20 @@ private InetSocketAddress nextNameServerAddress() {
             InetSocketAddress nameServerAddr, DnsQuestion question) {
 
         return query0(checkNotNull(nameServerAddr, "nameServerAddr"),
+                checkNotNull(question, "question"),
```

Comment:
IMHO, all those `checkNotNull`s should be done once in `query0` instead of every caller.